packages/solidity-contracts/deploy/beta5devnet/001.chain_state.ts

```diff
@@ -0,0 +1,36 @@
+import type { HardhatRuntimeEnvironment } from 'hardhat/types';
+import type { DeployFunction } from 'hardhat-deploy/dist/types';
+
+import { FuelChainState__factory as FuelChainState } from '../../typechain';
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  const {
+    ethers,
+    upgrades: { deployProxy, erc1967 },
+    deployments: { save },
+  } = hre;
+  const [deployer] = await ethers.getSigners();
+
+  const { deployTransaction, address } = await deployProxy(
+    new FuelChainState(deployer),
+    [],
+    {
+      initializer: 'initialize',
+    }
+  );
+  await deployTransaction.wait();
+  const implementation = await erc1967.getImplementationAddress(address);
+
+  console.log('Deployed FuelChainState at', address);
+  await save('FuelChainState', {
+    address,
+    abi: [],
+    implementation,
+  });
+
+  return true;
```

Comment:
The returns are used as failsafes so that you don't accidentally rerun a script. Controls migrations (ie updates) of the smart contracts. This way you cannot do things like downgrade or redeploy a smart contract.